### PR TITLE
Feature: Add booking report module

### PR DESCRIPTION
**Pull request by proxiex**
- allows a user to get flight booking report history via email
Screenshot 2019-06-22 at 11 00 21 AM

### CODE REVIEW

File: flights/background_jobs.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                booking.flight.arrival_time,
+                booking.flight.price
+            ])
+
+    email_booking_report(user, start_date, end_date, file_name)
+
+
+@api_view(['GET'])
+@permission_classes((IsAuthenticated, ))
+def report(request):
```
Comment (tonymontaro) [inline flights/background_jobs.py]: This function-based views should be within your views.py file (as is convention and makes your code easier to follow). You can import `bg_job` as needed.

File: flights/helpers/decorators.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from threading import Thread
+from functools import wraps
+
+
+def start_backgroun_job(func):
```
Comment (tonymontaro) [inline flights/helpers/decorators.py:5]: Let's give this function a doc-string. 
I recommend adding an additional tool to help with this; `pip install flake8-docstrings` (https://pypi.org/project/flake8-docstrings/).

Run one last `flake8` validation for the project and fix any reported voilations.

**Review by tonymontaro — CHANGES_REQUESTED** [review]
Excellent work! You're mostly done with the backend. Just some final checks and we're good.

File: flights/background_jobs.py (hunk 1 above)
Comment (proxiex) [reply flights/background_jobs.py]: All done

File: flights/helpers/decorators.py (hunk 2 above)
Comment (proxiex) [reply flights/helpers/decorators.py:5]: all done

**Review by tonymontaro — APPROVED** [review]

**Pull request merged**